### PR TITLE
[fix](RLE) Quit when the RLE length repeat count is larger than int32_t

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by ByteYue** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Review by Yukang-Lian — APPROVED** [review]

File: be/src/util/rle_encoding.h [hunk 1]
```diff
@@ -493,7 +493,17 @@ template <typename T>
...
+    // Avoid attempting to manage run lengths exceeding the capacity of an int32_t.
+    // Note that the Parquet standard prohibits longer runs - refer to PARQUET-1290 for details.
+    if (repeat_count_ > std::numeric_limits<int32_t>::max()) [[unlikely]] {
+        num_buffered_values_ = 0;
+        repeat_count_ = 0;
+        LOG(WARNING) << "Run length exceeds int32_t";
+        return;
+    }
+
+    uint32_t indicator_value = repeat_count_ << 1 | 0;
```
Comment (gavinchou) [inline be/src/util/rle_encoding.h:506]: why "| 0" ?

File: be/src/util/rle_encoding.h [hunk 2]
```diff
@@ -493,7 +493,17 @@ template <typename T>
...
     DCHECK_GT(repeat_count_, 0);
     // The lsb of 0 indicates this is a repeated run
-    int32_t indicator_value = repeat_count_ << 1 | 0;
+
+    // Avoid attempting to manage run lengths exceeding the capacity of an int32_t.
+    // Note that the Parquet standard prohibits longer runs - refer to PARQUET-1290 for details.
+    if (repeat_count_ > std::numeric_limits<int32_t>::max()) [[unlikely]] {
+        num_buffered_values_ = 0;
+        repeat_count_ = 0;
+        LOG(WARNING) << "Run length exceeds int32_t";
```
Comment (gavinchou) [inline be/src/util/rle_encoding.h:502]: also print repeat_count_

**Comment by dataroaring** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Pull request closed without merging**